eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPublisher.scala

```diff
@@ -18,442 +18,294 @@ package fr.acinq.eclair.channel.publish
-      case Stop => Behaviors.stopped
+      case Stop =>
+        // We can't stop right now, the child actor is currently funding the transaction and will send its result soon.
+        // We just wait for the funding process to finish before stopping (in the next state).
+        timers.startSingleTimer(Stop, 1 second)
+        Behaviors.same
     }
   }
 
-  def checkTimeLocks(replyTo: ActorRef[TxPublisher.PublishTxResult], cmd: TxPublisher.PublishReplaceableTx, htlcTxWithWitnessData: HtlcTxAndWitnessData, targetFeerate: FeeratePerKw): Behavior[Command] = {
-    val timeLocksChecker = context.spawn(TxTimeLocksMonitor(nodeParams, watcher, loggingInfo), "time-locks-monitor")
-    timeLocksChecker ! CheckTx(context.messageAdapter[TxTimeLocksMonitor.TimeLocksOk](_ => TimeLocksOk), cmd.txInfo.tx, cmd.desc)
+  // Wait for our transaction to be confirmed or rejected from the mempool.
+  // If we get close to the confirmation target and our transaction is stuck in the mempool, we will initiate an RBF attempt.
+  def wait(txMonitor: ActorRef[MempoolTxMonitor.Command], tx: FundedTx): Behavior[Command] = {
     Behaviors.receiveMessagePartial {
-      case TimeLocksOk =>
-        val htlcFeerate = cmd.commitments.localCommit.spec.htlcTxFeerate(cmd.commitments.commitmentFormat)
-        if (targetFeerate <= htlcFeerate) {
-          val channelKeyPath = keyManager.keyPath(cmd.commitments.localParams, cmd.commitments.channelConfig)
-          val localPerCommitmentPoint = keyManager.commitmentPoint(channelKeyPath, cmd.commitments.localCommit.index)
-          val localHtlcBasepoint = keyManager.htlcPoint(channelKeyPath)
-          val localSig = keyManager.sign(htlcTxWithWitnessData.txInfo, localHtlcBasepoint, localPerCommitmentPoint, TxOwner.Local, cmd.commitments.commitmentFormat)
-          val signedHtlcTx = htlcTxWithWitnessData.addSigs(localSig, cmd.commitments.commitmentFormat)
-          log.info("publishing {} without adding inputs: txid={}", cmd.desc, signedHtlcTx.tx.txid)
-          publish(replyTo, cmd, signedHtlcTx.tx, signedHtlcTx.fee)
+      case WrappedTxResult(txResult) =>
+        txResult match {
+          case MempoolTxMonitor.TxInMempool(_, currentBlockCount) =>
+            // We avoid a herd effect whenever we fee bump transactions.
+            timers.startSingleTimer(CheckFeeKey, CheckFee(currentBlockCount), (1 + Random.nextLong(nodeParams.maxTxPublishRetryDelay.toMillis)).millis)
+            Behaviors.same
+          case MempoolTxMonitor.TxRecentlyConfirmed(_, _) => Behaviors.same // just wait for the tx to be deeply buried
+          case MempoolTxMonitor.TxDeeplyBuried(confirmedTx) => sendResult(TxPublisher.TxConfirmed(cmd, confirmedTx))
+          case MempoolTxMonitor.TxRejected(_, reason) =>
+            replyTo ! TxPublisher.TxRejected(loggingInfo.id, cmd, reason)
+            // We wait for our parent to stop us: when that happens we will unlock utxos.
```

Comment:
Because in failure cases we need to unlock utxos (we don't in success cases).
We could start unlocking right now though, I'll revisit that.

Reply:
This is now more intuitive in https://github.com/ACINQ/eclair/pull/2113/commits/25fa5e60396c16df77f2a0fedb431d283c4a54ca

Reply:
I still think we could have more symmetry between the success and failure cases (specifically it bother me a little bit that we are not using `sendResult` in this error case). Maybe we should call `abandonTransaction`/`unlockOutpoints` from `sendResult` in the non-sucess case?

Btw, you haven't explicitly handled failures of `abandonTransaction`/`unlockOutpoints` AFAICT, is it by design? We don't want silent failures here do we? We could instead define `UnlockUtxos` as follows and print an error message if the embedded `Try` is a failure:

```scala
case class UnlockUtxosResult(try: Try[...])
```

Reply:
> Maybe we should call abandonTransaction/unlockOutpoints from sendResult in the non-success case?

We could do that, I'll try it.

> Btw, you haven't explicitly handled failures of abandonTransaction/unlockOutpoints AFAICT, is it by design? We don't want silent failures here do we?

We really want silent errors here, this call will fail in many cases (for example when we do it for both transactions when we're stopped in the middle of processing), but it's ok.

The behavior we want is "abandon the transaction and unlock the inputs if that's possible, otherwise we're all good".
I think that explicitly handling the errors would be unnecessary code, it would make it feel like we should implement checks that would be redundant with the ones inside bitcoind.

Reply:
> Maybe we should call abandonTransaction/unlockOutpoints from sendResult in the non-success case?

Done in https://github.com/ACINQ/eclair/pull/2113/commits/315b0a63b7e794504ac8908181ea680b4018161f
It's better indeed, we have a single place where we send the response to our parent :+1:

Reply:
Fair enough